src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,23 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Atomically appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        if (false !== $originalContent = @file_get_contents($filename)) {
```

Comment:
In fact, using `LOCK_EX` doesn't make `file_put_contents` atomic, so in both cases (using `file_get_contents/file_put_contents` or `file_put_contents` with flags) the append would not be atomic. Considering this I prefer to let `dumpFile()` as it is since the behavior is better (maybe not really atomic, but more than using `file_put_contents` directly).
Refs:
https://github.com/symfony/symfony/issues/10018#issuecomment-68598616
http://stackoverflow.com/questions/4899737/should-lock-ex-on-both-read-write-be-atomic

About `appendToFile`, I'm still not sure what is better between using `file_get_contents` or `file_put_contents` with FILE_APPEND, but I tend to prefer using `file_get_contents` to avoid having `file_put_contents` breaks in the middle of the write..